### PR TITLE
Update Bitcoin Core node version

### PR DESCRIPTION
**Pull request by lontivero**
This PR updates the Bitcoin Core node version in order to be able to start implementing the new `getfilter` RPC call for Wasabi usage. This is a WIP

### CODE REVIEW

**Comment by nopara73** [comment]
> in order to be able to start implementing the new getfilter RPC call for Wasabi usage. 

How do you imagine that?

**Comment by lontivero** [comment]
Something like

```c#
	public CompactFilter GetFilter(uint256 blockHash)
	{
		var resp = SendCommand("getfilter", blockHash, "basic");
		return ParseCompactFilter(resp);
	}

	public async Task<CompactFilter> GetFilterAsync(uint256 blockHash)
	{
		var resp = await SendCommandAsync("getfilter", blockHash, "basic").ConfigureAwait(false);
		return ParseCompactFilter(resp);
	}

	public class CompactFilter
	{
		public GolombRiceFilter Filter { get; }
		public uint256 Header { get; }
	}
```

**Comment by nopara73** [comment]
I mean how do you imagine using it in Wasabi? Our filters are 200MB, these filters are 4GB.  

Btw, I had an idea. I think it can come really handy for Bitcoin Core integration. Our last large problem is porting the backend filter building into the client, but with this we don't need to. Since we already only iterate through the last 100 filters, except at new wallet generation, we could make a great use for this. We don't need to build out filters on the client side, but it's ok to let Core build their filters, and we could create 2 IIndexProviders, where the new one would be using RPC directly.

**Comment by nopara73** [comment]
Regarding this PR. Tests fail.

**Comment by lontivero** [comment]
Yes, I know. A lot of RPC methods changed and have different parameter or return different results now. I am on that right now.

**Comment by lontivero** [comment]
I will continue tomorrow. I don't know why `CanTestMempoolAccept` fails. Also, I have to see why the `AbortScanTxoutSetAsync` method fails now, it seems it requires some extra parameter maybe.

**Comment by lontivero** [comment]
RPC is updated now and all the RPC tests pass but these other two tests fail:

* NBitcoin.Tests.ProtocolTests.CanExchangeFastPingPong
* NBitcoin.Tests.ProtocolTests.CanGetMerkleRoot

I don't know why. @NicolasDorier any advice would be great.

**Comment by lontivero** [comment]
**It is ready for review**. The test failing has nothing to do with this work and passed all the previous commits, i mean, it is a random error.

**Comment by knocte** [comment]
> so if the user uses older Core, then it'll fail

how about checking the version of Core via RPC and fail first if older version before trying to do anything

**Comment by NicolasDorier** [comment]
hey sorry I am back.

> There are backwards incompatible changes, so if the user uses older Core, then it'll fail, but I guess there's nothing to do about that, Bitcoin Core just cannot keep not breaking the user space.

This.

The `RPCClient` should try to silently fix breaking changes from core as much as possible, I don't see your PR doing this.

**Comment by NicolasDorier** [comment]
@knocte 

> how about checking the version of Core via RPC and fail first if older version before trying to do anything

For some stuff it is not really needed, you can just observe the shape of the json.
For other stuff, this is what the `ScanCapabilities` in the RPCClient call is about.

**Comment by lontivero** [comment]
@NicolasDorier let me know if this version is better.

File: NBitcoin/RPC/RPCClient.cs [hunk 1]
```diff
@@ -1389,9 +1402,21 @@ public MempoolAcceptResult TestMempoolAccept(Transaction transaction, bool allow
 			return TestMempoolAcceptAsync(transaction, allowHighFees).GetAwaiter().GetResult();
 		}
 
-		public async Task<MempoolAcceptResult> TestMempoolAcceptAsync(Transaction transaction, bool allowHighFees = false)
+        public async Task<MempoolAcceptResult> TestMempoolAcceptAsync(Transaction transaction, bool allowHighFees = false)
```
Comment (NicolasDorier) [inline NBitcoin/RPC/RPCClient.cs]: should be tab

File: NBitcoin.Tests/NodeBuilderEx.cs [hunk 2]
```diff
@@ -47,7 +47,8 @@ public static NodeBuilder Create([CallerMemberName] string caller = null)
 			//var builder = NodeBuilder.Create(NodeDownloadData.Dystem.v1_0_9_9, Altcoins.Dystem.Instance.Regtest, caller);
 			//var builder = NodeBuilder.Create(NodeDownloadData.Bitcoinplus.v2_7_0, Altcoins.AltNetworkSets.Bitcoinplus.Regtest, caller);
 
-			var builder = NodeBuilder.Create(NodeDownloadData.Bitcoin.v0_18_1, Altcoins.AltNetworkSets.Bitcoin.Regtest, caller);
+			var builder = NodeBuilder.Create(NodeDownloadData.Bitcoin.v0_19_0_1, Altcoins.AltNetworkSets.Bitcoin.Regtest, caller);
+			//var builder = NodeBuilder.Create(NodeDownloadData.Bitcoin.v0_18_1, Altcoins.AltNetworkSets.Bitcoin.Regtest, caller);
```
Comment (NicolasDorier) [inline NBitcoin.Tests/NodeBuilderEx.cs]: remove this line

**Comment by NicolasDorier** [comment]
This is better, can you fix the two smalls nits I pointed out and rebase? I should have fixed the unit tests, sorry for that. Will make a release once merged this.

**Comment by NicolasDorier** [comment]
Sorry, fixed the build again... can you rebase?

**Pull request merged**